flink-staging/flink-ml/src/test/scala/org/apache/flink/ml/math/SparseVectorSuite.scala

```diff
@@ -87,4 +87,33 @@ class SparseVectorSuite extends FlatSpec with Matchers {
 
     sparseVector should not equal(copy)
   }
+
+  it should "calculate dot product with SparseVector" in {
+    val vec1 = SparseVector.fromCOO(4, (0, 1), (1, 0), (2, 1), (3, 0))
+    val vec2 = SparseVector.fromCOO(4, (0, 0), (1, 1), (2, 0), (3, 1))
```

Comment:
Again non-zero entries are continuous starting from index 0. We should add a test case which works on more irregular non-zero entries.